apps/site/lib/site/content_rewriters/liquid_objects/fare.ex

```diff
@@ -185,18 +220,33 @@ defmodule Site.ContentRewriters.LiquidObjects.Fare do
     end
   end
 
-  defp compose_args(invalid_error), do: invalid_error
+  defp compose_args(invalid_error) do
+    invalid_error
+  end
 
-  @spec request_fares({:ok, [repo_arg] | {summary_mode, [repo_arg]}} | request_error) ::
-          [Summary.t()] | [Fare.t()] | request_error
+  @spec request_fares(request_tuple | request_error) :: [Summary.t()] | [Fare.t()] | request_error
+  # If the mode indicates a summary will be returned, format the results of get_fares/1 as a summary
   defp request_fares({:ok, {mode, args}}) when mode in @summary_atoms do
     args
     |> get_fares()
     |> Format.summarize(mode)
   end
 
-  defp request_fares({:ok, args}), do: get_fares(args)
-  defp request_fares(error), do: error
+  # If the request is for a zone or interzone fare, merge placeholder keys into tuple value for :name
+  defp request_fares({:ok, {zone_type, zone_id, args}}) when zone_id in @zone_id do
+    args
+    |> Keyword.put(:name, {zone_type, zone_id})
```

Comment:
why does this need to happen here, instead of inside of `compose_args`?

Reply:
You're right that it should be in `compose_args`. I had it separate but it really didn't need to be.